readers/influxdb/messages.go

```diff
@@ -1,150 +1,186 @@
+	sb.WriteString(`import "influxdata/influxdb/v1"`)
+	sb.WriteString(fmt.Sprintf(`from(bucket: "%s")`, repo.cfg.Bucket))
+	// FluxQL syntax requires timeRange filter in this position, do not change.
+	sb.WriteString(timeRange)
+	// This is required to get messsage structure. Otherwise query returns fields in seperate rows.
+	sb.WriteString(`|> v1.fieldsAsCols()`)
+	sb.WriteString(`|> group()`)
+	sb.WriteString(fmt.Sprintf(`|> filter(fn: (r) => r._measurement == "%s")`, format))
+	sb.WriteString(condition)
+	sb.WriteString(`|> sort(columns: ["_time"], desc: true)`)
+	sb.WriteString(fmt.Sprintf(`|> limit(n:%d,offset:%d)`, rpm.Limit, rpm.Offset))
+	sb.WriteString(`|> yield(name: "sort")`)
+	query := sb.String()
+	resp, err := queryAPI.Query(context.Background(), query)
 	if err != nil {
 		return readers.MessagesPage{}, errors.Wrap(readers.ErrReadMessages, err)
 	}
-	if resp.Error() != nil {
-		return readers.MessagesPage{}, errors.Wrap(readers.ErrReadMessages, resp.Error())
-	}
-
-	if len(resp.Results) == 0 || len(resp.Results[0].Series) == 0 {
-		return readers.MessagesPage{}, nil
-	}
-
+	/*
+		if len(resp.Results) == 0 || len(resp.Results[0].Series) == 0 {
+			return readers.MessagesPage{}, nil
+		}
+	*/
 	var messages []readers.Message
-	result := resp.Results[0].Series[0]
-	for _, v := range result.Values {
-		msg, err := parseMessage(format, result.Columns, v)
+
+	var valueMap map[string]interface{}
+	for resp.Next() {
+		valueMap = resp.Record().Values()
+		msg, err := parseMessage(format, valueMap)
 		if err != nil {
 			return readers.MessagesPage{}, err
 		}
 		messages = append(messages, msg)
 	}
+	if resp.Err() != nil {
+		return readers.MessagesPage{}, errors.Wrap(readers.ErrReadMessages, resp.Err())
+	}
 
-	total, err := repo.count(format, condition)
+	fmt.Println("Query: ", query)
+
+	total, err := repo.count(format, condition, timeRange)
 	if err != nil {
 		return readers.MessagesPage{}, errors.Wrap(readers.ErrReadMessages, err)
 	}
 
 	page := readers.MessagesPage{
 		PageMetadata: rpm,
-		Total:        total,
+		Total:        total, //total,
 		Messages:     messages,
 	}
 
 	return page, nil
+
 }
 
-func (repo *influxRepository) count(measurement, condition string) (uint64, error) {
-	cmd := fmt.Sprintf(`SELECT COUNT(*) FROM %s WHERE %s`, measurement, condition)
-	q := influxdata.Query{
-		Command:  cmd,
-		Database: repo.database,
-	}
+func (repo *influxRepository) count(measurement, condition string, timeRange string) (uint64, error) {
+
+	var sb strings.Builder
+	sb.WriteString(`import "influxdata/influxdb/v1"`)
+	sb.WriteString(fmt.Sprintf(`from(bucket: "%s")`, repo.cfg.Bucket))
+	sb.WriteString(timeRange)
+	sb.WriteString(`|> v1.fieldsAsCols()`)
+	sb.WriteString(fmt.Sprintf(`|> filter(fn: (r) => r._measurement == "%s")`, measurement))
+	sb.WriteString(condition)
+	sb.WriteString(`|> group()`)
+	sb.WriteString(`|> count(column:"_measurement")`)
+	sb.WriteString(`|> yield(name: "count")`)
```

Comment:
Same here - try to simplify. Subsequent `sb.WriteString` calls with a simple string as an argument do not make much sense.

Reply:
You are right sorry. Changed both string builders to `fmt.sprintf()`